File: services/chatbot/internal/models/rule.go
package models

import "time"

type Rule struct {
	ID                int       `json:"id" db:"id"`
	Response          string    `json:"response" db:"response"`
	IsActive          bool      `json:"is_active" db:"is_active"`
	Priority          int       `json:"priority" db:"priority"`
	MatchType         string    `json:"match_type" db:"match_type"`
	CreatedAt         time.Time `json:"created_at" db:"created_at"`
	UpdatedAt         time.Time `json:"updated_at" db:"updated_at"`
	RuleName          string    `json:"rule_name" db:"rule_name"`
	HideAfterReply    bool      `json:"hide_after_reply" db:"hide_after_reply"`
	SendToInbox       bool      `json:"send_to_inbox" db:"send_to_inbox"`
	InboxMessage      string    `json:"inbox_message" db:"inbox_message"`
	InboxImage        string    `json:"inbox_image" db:"inbox_image"`
	CreatedBy         *int      `json:"created_by" db:"created_by"`
	Keywords          []string  `json:"keywords" db:"-"` // not a DB column, loaded from rule_keywords
}

type RuleResponse struct {
	ID           int       `json:"id" db:"id"`
	RuleID       int       `json:"rule_id" db:"rule_id"`
	ResponseText string    `json:"response_text" db:"response_text"`
	ResponseType string    `json:"response_type" db:"response_type"`
	MediaURL     *string   `json:"media_url,omitempty" db:"media_url"`
	Weight       int       `json:"weight" db:"weight"`
	IsActive     bool      `json:"is_active" db:"is_active"`
	CreatedAt    time.Time `json:"created_at" db:"created_at"`
}

File: web/app/types/rule.ts
export type Response = {
  text: string
  image?: string // URL or base64 image for this specific response
}

export type ConnectedPage = {
  id: string
  name: string
  pageId: string
  connected: boolean
  enabled: boolean
}

export type RulePair = {
  id?: string // Add unique ID for each pair
  title?: string // Custom title for each pair 
  hasManuallyEditedTitle?: boolean // Track if user manually edited title
  rules: string[]
  responses: Response[]
  hideCommentsAfterReply?: boolean
  enableInboxIntegration?: boolean
  inboxResponse?: string
  inboxImage?: string
  enabled?: boolean
  expanded?: boolean
  selectedPages?: string[]
}

export type FilterSettings = {
  skipMentions: boolean
  skipStickers: boolean
}

export type FallbackSettings = {
  enabled: boolean
  responses: Response[]
  selectedPages?: string[]
  hideAfterReply?: boolean
  sendToInbox?: boolean
  inboxMessage?: string
  inboxImage?: string
}

export type Settings = {
  defaultResponses: string[]
  enableDefault: boolean
  noTag: boolean
  noSticker: boolean
  hideCommentsAfterReply: boolean
  enableInboxIntegration: boolean
  inboxResponse: string
  inboxImage?: string
  filterSettings: FilterSettings
  fallbackSettings: FallbackSettings
}

export type RuleData = {
  rules: RulePair[]
} & Settings

export type Rule = {
  id: number
  keywords: string[]
  response: string
  is_active: boolean
  priority: number
  match_type: string
  created_at: string
  updated_at: string
  rule_name: string
  hide_after_reply: boolean
  send_to_inbox: boolean
  inbox_message: string
  inbox_image: string
  created_by?: number | null
}

export type RuleResponse = {
  pairs: Rule[]
  total?: number
  success?: boolean
}

export type ApiResponse<T = any> = {
  success: boolean
  data?: T
  error?: string
  message?: string
  timestamp?: string
}
